### PR TITLE
Add io.github.sigmasd.IRust

### PR DESCRIPTION
**Pull request by sigmaSd**
Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements](https://docs.flathub.org/docs/for-app-authors/requirements) and [App Maintenance](https://docs.flathub.org/docs/for-app-authors/maintanance) pages.
- [x] My pull request follows the instructions at [App Submission](https://docs.flathub.org/docs/for-app-authors/submission).
- [x] I am using only the minimal set of permissions. (If not, please explain each non-standard permission.)
- [x] All assets referenced in the manifest are redistributable by any party. If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. Link:
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. (If not, explain why.)

### CODE REVIEW

**Comment by bbhtt** [comment]
Please answer the PR template, it is required.

File: io.github.sigmasd.IRust.desktop
Comment (bbhtt) [inline io.github.sigmasd.IRust.desktop:1]: This should be upstream

File: io.github.sigmasd.IRust.metainfo.xml
Comment (bbhtt) [inline io.github.sigmasd.IRust.metainfo.xml:1]: This should be upstream

File: io.github.sigmasd.IRust.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="console-application">
+  <id>io.github.sigmasd.IRust</id>
+  <name>IRust</name>
+  <summary>Rust Repl</summary>
+
+  <metadata_license>MIT</metadata_license>
+  <project_license>MIT</project_license>
```
Comment (bbhtt) [inline io.github.sigmasd.IRust.metainfo.xml:8]: Add developer_name

File: io.github.sigmasd.IRust.svg
Comment (bbhtt) [inline io.github.sigmasd.IRust.svg:1]: This should be upstream

File: io.github.sigmasd.IRust.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+app-id: io.github.sigmasd.IRust
+
+runtime: org.freedesktop.Sdk
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.rust-stable
+
+command: irust
+
```
Comment (bbhtt) [inline io.github.sigmasd.IRust.yml:10]: finish-args is required

File: io.github.sigmasd.IRust.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      IRust is a repl for the rust language.
+    </p>
+  </description>
+
+  <launchable type="desktop-id">IRust.desktop</launchable>
+
+  <screenshots>
+    <screenshot type="default">
+      <caption>IRust screenshot</caption>
+      <image>https://github.com/sigmaSd/IRust/blob/13e0f71ec4c3e765ecff3a2e6d1ed900ea0b69f9/irust.png</image>
```
Comment (bbhtt) [inline io.github.sigmasd.IRust.metainfo.xml:21]:
```suggestion
      <image>https://raw.githubusercontent.com/sigmaSd/IRust/13e0f71ec4c3e765ecff3a2e6d1ed900ea0b69f9/irust.png</image>
```

The direct link is needed:

```
appstream-util validate io.github.sigmasd.IRust.metainfo.xml 
io.github.sigmasd.IRust.metainfo.xml: FAILED:
• file-invalid          : <screenshot> failed to load [https://github.com/sigmaSd/IRust/blob/13e0f71ec4c3e765ecff3a2e6d1ed900ea0b69f9/irust.png]
Validation of files failed
```

File: io.github.sigmasd.IRust.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  <metadata_license>MIT</metadata_license>
+  <project_license>MIT</project_license>
+
+  <description>
+    <p>
+      IRust is a repl for the rust language.
+    </p>
+  </description>
+
+  <launchable type="desktop-id">IRust.desktop</launchable>
```
Comment (bbhtt) [inline io.github.sigmasd.IRust.metainfo.xml:16]:
```suggestion
  <launchable type="desktop-id">io.github.sigmasd.IRust.desktop</launchable>
```

**Comment by bbhtt** [comment]
```
rustfmt is not installed, it's required for beautifying repl code
Do you want IRust to install it? [Y/n]: 
y
rustup is not installed.
rustup is required to install rustfmt
error while installing rustfmt
```

**Comment by sigmaSd** [comment]
I'm not sure about the options dependences, I thought I'm going to remove the welcome screen for the flatpak, but maybe it's not that much work to also include the optional dependences

File: io.github.sigmasd.IRust.yml (hunk 2 above)
Comment (hfiguiere) [reply io.github.sigmasd.IRust.yml:10]: it seems to be terminal only.

**Comment by sigmaSd** [comment]
I addressed all comments, can you please take a second look



> You need to add rustup to the manifest
> 
> ```
> rustfmt is not installed, it's required for beautifying repl code
> Do you want IRust to install it? [Y/n]: 
> y
> rustup is not installed.
> rustup is required to install rustfmt
> error while installing rustfmt
> ```

I disabled the scan for optional dependencies  for now on the flatpak package, I can add those on own my pace later.

**Comment by hfiguiere** [comment]
bot, build io.github.sigmasd.IRust

File: io.github.sigmasd.IRust.yml (hunk 2 above)
Comment (hfiguiere) [reply io.github.sigmasd.IRust.yml:10]:
```suggestion
finish-args: []
```

This should be enough to make the linter happy.

CLI application aren't really the ideal use case for flatpak.

Comment (sigmaSd) [reply io.github.sigmasd.IRust.yml:10]: I understand that, my motivation is mostly the exposure the tools get through store, I discovered a lot of cool projects through gnome software.

File: io.github.sigmasd.IRust.yml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  - org.freedesktop.Sdk.Extension.rust-stable
+
+command: irust
+finish-args: []
+modules:
+  - name: irust
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 $(which cargo) -t /app/bin/
+      - install -Dm755 $(which rustc) -t /app/bin/
```
[the remark this answers is not in the document]
Comment (sigmaSd) [reply io.github.sigmasd.IRust.yml:16]: I'm copying from the rust stable SDK, my program should continue to work with the recent versions of stable no problem. I see what you mean about controlling the exact version but in practice I can forsee that bundling cargo is going to be a pain , and there is no tangible benefit in my opinion.

**Comment by hfiguiere** [comment]
bot, build io.github.sigmasd.IRust

**Comment by sigmaSd** [comment]
Should I keep pushing updates (mostly to update the version of irust) or should I wait for this pr to be accepted ?

**Pull request closed without merging**